Resources/public/js/views/dashboard/ez-dashboardblockallcontentview.js

```diff
@@ -0,0 +1,157 @@
+        },
+
+        /**
+         * Renders the dashboard block view
+         *
+         * @method render
+         * @return {eZ.DashboardBlockAllContentView} the view itself
+         */
+        render: function () {
+            this.get('container').setHTML(this.template()).addClass(CLASS_LOADING);
+
+            return this;
+        },
+
+        /**
+         * Shows the row options
+         *
+         * @method _uiShowOptions
+         * @protected
+         * @param event {Object} event facade
+         */
+        _uiShowRowOptions: function (event) {
+            this._uiHideRowOptions();
+
+            event.currentTarget.addClass(CLASS_ROW_SELECTED);
+
+            this._clickOutsideHandler = this.get('container').one(SELECTOR_CONTENT).on('clickoutside', Y.bind(this._uiHideRowOptions, this));
+        },
+
+        /**
+         * Clears the click outside handler
+         *
+         * @method _clearClickOutsideHandler
+         * @protected
+         */
+        _clearClickOutsideHandler: function () {
+            if (this._clickOutsideHandler) {
+                this._clickOutsideHandler.detach();
+            }
+        },
+
+        /**
+         * Hides the row options
+         *
+         * @method _uiHideOptions
+         * @protected
+         */
+        _uiHideRowOptions: function () {
+            this.get('container').all(SELECTOR_ROW).removeClass(CLASS_ROW_SELECTED);
+
+            this._clearClickOutsideHandler();
+
+        },
+
+        /**
+         * Makes request for data
+         *
+         * @method _getContentData
+         * @protected
+         * @param event {Object} event facade
+         */
+        _getContentData: function () {
+            this.fire('contentSearch', {
+                viewName: 'all-content',
+                loadContentType: true,
+                search: {
+                    criteria: {SubtreeCriterion: this.get('rootLocation').get('pathString')},
+                    /*
+                     * @TODO sort items by modification date
+                     * see https://jira.ez.no/browse/EZP-24998
+                     *
+                     * sortClauses: {DateModifiedClause: 'DESC'},
+                     */
+                    limit: 10
+                },
+                callback: Y.bind(this._renderContentRows, this)
+            });
+        },
+
+        /**
+         * Renders content rows in a block
+         *
+         * @method _renderContentRows
+         * @protected
+         * @param error {Boolean} is error?
+         * @param list {Array} rows content data
+         */
+        _renderContentRows: function (error, list) {
+            var rows = list.map(function (data) {
+                var content = data.content.toJSON(),
+                    contentTypeName = data.contentType.get('names')[content.mainLanguageCode];
+
+                return {
+                    title: content.name,
```

Comment:
you got me wrong in https://github.com/ezsystems/PlatformUIBundle/pull/589#discussion_r64001614, the point of calling `toJSON` is not to avoid calling `get` multiple times, it's to provide a full representation of the object in the template. That way, integrators of eZ Platform can customize the block just by using another template (for instance to add new column for another property we don't display by default). So as I was writing in the initial comment, here you should have something like (without the comments of course):
```js
return {
    location: location.toJSON(), // because of https://github.com/ezsystems/PlatformUIBundle/pull/589/files#r65134592
    contentInfo: location.get('contentInfo').toJSON(), // to ease the template a bit
    contentType: contentType.toJSON(),
};
```